### PR TITLE
Exclude cdacreader from mono packs

### LINKED ISSUE #103486: [Mono] Build fails with libcdacreader.so missing from manifest
### Description

When attempting to build Mono for linux-x64 using the build command `build.sh -ci -arch x64 -os linux -cross -s mono+clr.iltools+clr.hosts+libs+host+packs -c Release`, the build fails with this error:
```
/__w/1/s/.packages/microsoft.dotnet.sharedframework.sdk/9.0.0-beta.24311.10/targets/sharedfx.targets(296,5): error : The following files are missing entries in the templated manifest: [/__w/1/s/src/installer/pkg/sfx/Microsoft.NETCore.App/Microsoft.NETCore.App.Runtime.sfxproj]
/__w/1/s/.packages/microsoft.dotnet.sharedframework.sdk/9.0.0-beta.24311.10/targets/sharedfx.targets(296,5): error : libcdacreader.so. Add these file names with extensions to the 'PlatformManifestFileEntry' item group for the runtime pack and corresponding ref pack to include them in the platform manifest. [/__w/1/s/src/installer/pkg/sfx/Microsoft.NETCore.App/Microsoft.NETCore.App.Runtime.sfxproj]
```

Example pipeline with the failure: https://dev.azure.com/dnceng/internal/_build/results?buildId=2472669&view=results
This seems like a similar error as was hit here: https://github.com/dotnet/runtime/issues/67829.

### Reproduction Steps

Clone the runtime repo and build with `build.sh -ci -arch x64 -os linux -cross -s mono+clr.iltools+clr.hosts+libs+host+packs -c Release` using the build.sh.

### Expected behavior

Build succeeds

### Actual behavior

Build fails with:
```
/__w/1/s/.packages/microsoft.dotnet.sharedframework.sdk/9.0.0-beta.24311.10/targets/sharedfx.targets(296,5): error : The following files are missing entries in the templated manifest: [/__w/1/s/src/installer/pkg/sfx/Microsoft.NETCore.App/Microsoft.NETCore.App.Runtime.sfxproj]
/__w/1/s/.packages/microsoft.dotnet.sharedframework.sdk/9.0.0-beta.24311.10/targets/sharedfx.targets(296,5): error : libcdacreader.so. Add these file names with extensions to the 'PlatformManifestFileEntry' item group for the runtime pack and corresponding ref pack to include them in the platform manifest. [/__w/1/s/src/installer/pkg/sfx/Microsoft.NETCore.App/Microsoft.NETCore.App.Runtime.sfxproj]
```

### Regression?

This broke sometime Wednesday the 12th, with the diff between our last working and first broken build being: https://github.com/dotnet/runtime/compare/bb2a99b1e58311543ae26512e39ae626ba864faa...eb8f54d92b015c8b416f6e506fd4c3335740302c

### Known Workarounds

None

### Configuration

Build is completed with the latest runtime targeting linux-x64

### Other information

_No response_

### PR DESCRIPTION
**Pull request by am11**
Fix https://github.com/dotnet/runtime/issues/103486

### CODE REVIEW

File: src/installer/pkg/sfx/Microsoft.NETCore.App/Directory.Build.props [hunk 1]
```diff
@@ -120,6 +120,8 @@
     <PlatformManifestFileEntry Include="mscordaccore.dll" IsNative="true" />
     <PlatformManifestFileEntry Include="libmscordaccore.so" IsNative="true" />
     <PlatformManifestFileEntry Include="libmscordaccore.dylib" IsNative="true" />
+    <PlatformManifestFileEntry Include="libcdacreader.so" IsNative="true" />
```
Comment (am11) [inline src/installer/pkg/sfx/Microsoft.NETCore.App/Directory.Build.props:123]:
```suggestion
    <PlatformManifestFileEntry Include="cdacreader.dll" IsNative="true" />
    <PlatformManifestFileEntry Include="libcdacreader.so" IsNative="true" />
```

**Comment by lambdageek** [comment]
I don't think this what we want (yet).  We don't want to ship cdacreader yet https://github.com/dotnet/runtime/pull/100946#discussion_r1566320500

File: src/installer/pkg/sfx/Microsoft.NETCore.App/Directory.Build.props [hunk 2]
```diff
@@ -120,6 +120,9 @@
     <PlatformManifestFileEntry Include="mscordaccore.dll" IsNative="true" />
     <PlatformManifestFileEntry Include="libmscordaccore.so" IsNative="true" />
     <PlatformManifestFileEntry Include="libmscordaccore.dylib" IsNative="true" />
+    <PlatformManifestFileEntry Include="cdacreader.dll" IsNative="true" />
+    <PlatformManifestFileEntry Include="libcdacreader.so" IsNative="true" />
+    <PlatformManifestFileEntry Include="libcdacreader.dylib" IsNative="true" />
```
Comment (am11) [inline src/installer/pkg/sfx/Microsoft.NETCore.App/Directory.Build.props:125]:
```suggestion
    <PlatformManifestFileEntry Include="cdacreader.dll" ExcludeFromDataFiles="true" />
    <PlatformManifestFileEntry Include="libcdacreader.so" ExcludeFromDataFiles="true" />
    <PlatformManifestFileEntry Include="libcdacreader.dylib" ExcludeFromDataFiles="true" />
```

**Comment by am11** [comment]
@lambdageek fixed.

File: src/installer/pkg/sfx/Microsoft.NETCore.App/Directory.Build.props [hunk 3]
```diff
@@ -120,6 +120,9 @@
     <PlatformManifestFileEntry Include="mscordaccore.dll" IsNative="true" />
     <PlatformManifestFileEntry Include="libmscordaccore.so" IsNative="true" />
     <PlatformManifestFileEntry Include="libmscordaccore.dylib" IsNative="true" />
+    <PlatformManifestFileEntry Include="cdacreader.dll" ExcludeFromDataFiles="true" />
+    <PlatformManifestFileEntry Include="libcdacreader.so" ExcludeFromDataFiles="true" />
+    <PlatformManifestFileEntry Include="libcdacreader.dylib" ExcludeFromDataFiles="true" />
```
Comment (am11) [inline src/installer/pkg/sfx/Microsoft.NETCore.App/Directory.Build.props:125]:
```suggestion

    <!-- We don't want to ship cdacreader yet -->
    <PlatformManifestFileEntry Include="cdacreader.dll" ExcludeFromDataFiles="true" />
    <PlatformManifestFileEntry Include="libcdacreader.so" ExcludeFromDataFiles="true" />
    <PlatformManifestFileEntry Include="libcdacreader.dylib" ExcludeFromDataFiles="true" />

```

**Comment by lambdageek** [comment]
I'm having a hard time understanding what arcade is doing with `ExcludeFromDataFiles` over here

https://github.com/dotnet/arcade/blob/92a725aa4d9d6c13dc5229597b51b0b165b8b535/src/Microsoft.DotNet.SharedFramework.Sdk/targets/sharedfx.targets#L293

it looks like it's just silencing the error that the files aren't in the manifest.  But I don't understand if it's also excluding it from the runtime pack

**Comment by am11** [comment]
It is silencing the missing items which weren't included in the first place.

**Comment by lambdageek** [comment]
I think we need to exclude it here:

https://github.com/dotnet/runtime/blob/f8f450953d71dabe3fd01961c1d05b57c61e8faf/eng/liveBuilds.targets#L124

On CoreCLR RuntimeFiles gets populated from the sharedFramework subdirectory:

https://github.com/dotnet/runtime/blob/f8f450953d71dabe3fd01961c1d05b57c61e8faf/eng/liveBuilds.targets#L91

where we explicitly do _not_ put cdacreader yet:

https://github.com/dotnet/runtime/blob/f8f450953d71dabe3fd01961c1d05b57c61e8faf/src/native/managed/cdacreader/src/cdacreader.csproj#L20-L21

File: src/installer/pkg/sfx/Microsoft.NETCore.App/Directory.Build.props [hunk 4]
```diff
@@ -120,6 +120,12 @@
     <PlatformManifestFileEntry Include="mscordaccore.dll" IsNative="true" />
     <PlatformManifestFileEntry Include="libmscordaccore.so" IsNative="true" />
     <PlatformManifestFileEntry Include="libmscordaccore.dylib" IsNative="true" />
+
+    <!-- We don't want to ship cdacreader yet -->
+    <PlatformManifestFileEntry Include="cdacreader.dll" ExcludeFromDataFiles="true" />
+    <PlatformManifestFileEntry Include="libcdacreader.so" ExcludeFromDataFiles="true" />
+    <PlatformManifestFileEntry Include="libcdacreader.dylib" ExcludeFromDataFiles="true" />
+
```
Comment (am11) [inline src/installer/pkg/sfx/Microsoft.NETCore.App/Directory.Build.props:128]:
```suggestion
```

**Comment by am11** [comment]
@lambdageek this also fixes the build.

**Review by steveisok — APPROVED** [review]

**Comment by steveisok** [comment]
/ba-g Unblocks official build - specific to runtime pack generation.

**Comment by lambdageek** [comment]
Thank you for the quick fix.  I guess this was (probably one of) the reasons we had the RuntimeFlavor exclusion for cdacreader.  But this PR helps to clear up my confusion.

**Pull request merged**